sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2091,9 +2068,16 @@ device_image_plain ProgramManager::build(const device_image_plain &DeviceImage,
   auto CacheKey =
       std::make_pair(std::make_pair(std::move(SpecConsts), ImgId),
                      std::make_pair(PiDevice, CompileOpts + LinkOpts));
+
+  // CacheKey is captured by reference so when we overwrite it later we can
+  // reuse this function.
```

Comment:
Would it be helpful to modify the comment to point where such an overwrite is happening? Or would it become stale too soon?

Reply:
I worry it will be outdated too easily if we are specific here, but there is a comment with
> // Change device in the cache key to reduce copying of spec const data.

the place where the key is mutated.